File: problems/Invalid Transactions/invalidTransactions.js
/**
 * @param {string[]} transactions
 * @return {string[]}
 */
var invalidTransactions = function(transactions) {
    const result = [];
    const nameToTransactions = new Map();
    const store = new Set();
    const duplicates = new Map();

    for (const item of transactions) {
        const transaction = item.split(',');

        if (!duplicates.has(item)) {
            duplicates.set(item, 0);
        }
        duplicates.set(item, 1 + duplicates.get(item));

        if (1000 < parseInt(transaction[2])) {
            store.add(item);
        }

        if (nameToTransactions.has(transaction[0])) {
            const list = nameToTransactions.get(transaction[0]);

            for (let i = 0; list.length > i; ++i) {
                const sameNameTransaction = list[i].split(',');

                if (sameNameTransaction[3] !== transaction[3] &&
                    60 >= Math.abs(parseInt(sameNameTransaction[1]) - parseInt(transaction[1]))) {
                    store.add(list[i]);
                    store.add(item);
                }
            }

            nameToTransactions.get(transaction[0]).push(item);
        } else {
            nameToTransactions.set(transaction[0], [item]);
        }
    }

    for (const item of store) {
        for (let idx = 0; duplicates.get(item) > idx; ++idx) {
            result.push(item);
        }
    }

    return result;
};

File: problems/Invalid Transactions/invalidTransactions.java
class Solution {
    public List<String> invalidTransactions(String[] transactions) {
        List<String> result = new ArrayList<>();
        HashMap<String, Integer> duplicates = new HashMap<>();
        HashSet<String> store = new HashSet<>();
        HashMap<String, List<String>> nameToTransactions = new HashMap<>();

        for (String transaction : transactions) {
            String[] items = transaction.split(",");

            duplicates.put(transaction, 1 + duplicates.getOrDefault(transaction, 0));

            if (1000 < Integer.parseInt(items[2])) {
                store.add(transaction);
            }

            if (nameToTransactions.containsKey(items[0])) {
                List<String> list = nameToTransactions.get(items[0]);

                for (int idx = 0; list.size() > idx; ++idx) {
                    String[] sameNameTransaction = list.get(idx).split(",");

                    if (!items[3].equals(sameNameTransaction[3]) && 60 >= Math.abs(Integer.parseInt(items[1]) - Integer.parseInt(sameNameTransaction[1]))) {
                        store.add(transaction);
                        store.add(list.get(idx));
                    }
                }
            } else {
                nameToTransactions.put(items[0], new ArrayList<>());
            }

            nameToTransactions.get(items[0]).add(transaction);
        }

        for (String transaction : store) {
            for (int idx = 0; duplicates.get(transaction) > idx; ++idx) {
                result.add(transaction);
            }
        }

        return result;
    }
}
